api/v1alpha1/hostedcluster_types.go

```diff
@@ -118,6 +118,13 @@ const (
 	// removed when deleting the corresponding HostedCluster. If set to "true", resources created on the cloud provider during the life
 	// of the cluster will be removed, including image registry storage, ingress dns records, load balancers, and persistent storage.
 	CleanupCloudResourcesAnnotation = "hypershift.openshift.io/cleanup-cloud-resources"
+
+	// ResourceRequestOverrideAnnotationPrefix is a prefix for an annotation to override resource requests for a particular deployment/container
```

Comment:
I can see how the use case is valid for any consumer. I'd expect this to be an API field. No need to expose it right now though.